free-4.md

```diff
@@ -0,0 +1,80 @@
+
+### Why you?
+_What excites you about this idea? How did you come up with it?_
+
+As someone who doesn't own a small nursery of herbs, it's disappointing to find a particularly appetizing recipe, only to realize it requires some obscure herb (I'm looking at you tarragon). I'm also realizing more just how important some of these missing ingredient can be (e.g. I didn't add ginger to a Chinese dish and it tasted super bland). It would be nice to be able to just see what I can actually make given my ingredients rather than waste time looking at every possible chicken recipe.
+
+### Domain
+_Describe the project's domain in five words._
+
+Cooking with all your ingredients
+
+### Interface (syntax)
+_How might the user interact with the language? What does programming look 
+like? Why is this the right way to interact with the problem domain?_ 
+
+The user will provide the ingredients they have to the language and will receive a list of potential recipes. The user can also have additional fields that categorize the dishes, potentially by ethnic region, by allergies, among others. The program would probably have some request object that has several fields including ingredients and additioanl search query fields. 
+
+### Operation (semantics)
+_What might happen when a program runs? How does a program interact with the
+user? What kinds of errors might occur, and how might they be communicated to
+the user?_
+
+When a program runs, the user is given a truncated list of recipes they can make given their ingredients and potentially their categorization details. Some errors that might occur include no recipe existing in the intial database, improperly filled fields, misspelled ingredients, among others. It feels most errors will probably come form the use of natural language in the DSL.
+
+### Expressiveness
+_What should be easy to do in this language? What should be possible, but
+difficult? What should be impossible or very difficult?_
+
+The scope of this language is pretty limited, so it should be easy to find a recipe to cook for a meal, and difficult/very difficult/impossible to do much of anythign else.
+
+### Related work
+_Are there any other DSLs in this domain? If not, describe how you know there
+aren't and conjecture why not. If so, describe them and provide links. How well 
+do they address the need? Are there any particularly admirable qualities of the
+language? Are there parts of the language you think could be improved?_
+
+I didn't realize this before hand, but a quick Google-ing of the idea and [this](http://www.supercook.com/) comes up. It is similar, and seems to execute the idea rather well. I'm actually surprised this is not used more often as it seems like a pretty useful idea. It has a good database of existing recipes as well as user submitte recipes. The interface is a little slow and unresponsive though, which would be one area of improvement.
+
```

Comment:
I've used this before when I was feeling particularly uncreative. It's implemented fairly well but I like what you said about speeding up the recipe process. I also remember that I didn't like many of the recipe suggestions that were provided. You had a good idea about categories, especially if the user was in the mood to cook a dish from a particular ethnic background. In this language, it's annoying to try and remember everything that I have in the kitchen at a given time. One thing I would like to see in a language like this is something where the user could input some ingredients they have and maybe the language could prompt them by asking if they have some common ingredients (which might restrict them less in the types of results they get).